packages/backend/server/src/base/job/queue/def.ts

```diff
@@ -0,0 +1,69 @@
+import { applyDecorators, Logger, SetMetadata } from '@nestjs/common';
+
+import { CallMetric } from '../../metrics';
+
+declare global {
+  /**
+   * Job definitions can be extended by
+   *
+   * @example
+   *
+   * declare global {
+   *   interface Jobs {
+   *     'mail.newUserWelcome.send': {
+   *      ^^^ first segment must be namespace and a standalone queue will be created for each namespace
+   *       userId: User['id'];
+   *     }
+   *   }
+   * }
+   */
+  interface Jobs {}
+
+  type JobName = keyof Jobs;
+}
+
+export const JOB_METADATA = Symbol('JOB');
+export const JOB_NAMESPACES = new Set<string>();
+
+function JobHandlerWrapper(job: JobName): MethodDecorator {
+  // @ts-expect-error allow
+  return (
+    _target,
+    propertyKey,
+    desc: TypedPropertyDescriptor<(...args: any[]) => any>
+  ) => {
+    const originalMethod = desc.value;
+    if (!originalMethod) {
+      return desc;
+    }
+
+    desc.value = async function (...args: any[]) {
```

Comment:
make job handler run on the cls context https://papooch.github.io/nestjs-cls/features-and-use-cases/usage-outside-of-web-request

```ts
desc.value = async function (...args: any[]) {
   const cls = ClsServiceManager.getClsService()
   await cls.run(async () => {
     cls.set(CLS_ID, genRequestId('job'));
     new Logger('Job').log(`Job handler: ${job} (${propertyKey.toString()})`);
     return originalMethod.apply(this, args);
   });
});
```